src/framework/fixture.ts

```diff
@@ -43,11 +43,33 @@ export class Fixture {
     this.log('OK' + m);
   }
 
-  async shouldReject(p: Promise<unknown>, exceptionType?: string, msg?: string): Promise<void> {
+  protected async asyncExpectation(fn: () => Promise<void>): Promise<void> {
     this.numOutstandingAsyncExpectations++;
+    await fn();
+    this.numOutstandingAsyncExpectations--;
+  }
+
+  async shouldReject(p: Promise<unknown>, exceptionType?: string, msg?: string): Promise<void> {
```

Comment:
nit: If we don't care about the exceptionType, we'll have to call this as `shouldReject(p, undefined, msg)`
Thoughts on having the argument be a params dict `{ exceptionType?: string, msg?: string }` ?

Reply:
Decided to make the exception name required (and moved it to first position)